hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/OptUtils.java

```diff
@@ -188,6 +225,34 @@ private static boolean isPhysical(RelNode rel) {
         return rel.getTraitSet().getTrait(ConventionTraitDef.INSTANCE).equals(JetConventions.PHYSICAL);
     }
 
+    public static boolean isHazelcastTable(TableScan scan) {
+        HazelcastTable table = scan.getTable().unwrap(HazelcastTable.class);
+
+        return table != null;
+    }
+
+    public static HazelcastTable getHazelcastTable(TableScan scan) {
+        HazelcastTable table = scan.getTable().unwrap(HazelcastTable.class);
+
+        assert table != null;
+
+        return table;
+    }
+
+    public static HazelcastRelOptCluster getCluster(RelNode rel) {
+        assert rel.getCluster() instanceof HazelcastRelOptCluster;
+
+        return (HazelcastRelOptCluster) rel.getCluster();
+    }
+
+    public static DistributionTraitDef getDistributionDef(RelNode rel) {
```

Comment:
It's not really used.

Reply:
I forget to notice that to my mind we should keep all distribution trait classes. I am sure that we would support `ReplicatedMap` and that trait would be useful. 
But removal is also an option, since we have old codebase in separate branch.

Reply:
This particular method is not used. I don't have strong opinion about the trait classes really. They are not used but can be removed (if we decide so) in separate PR.